offer-manager-domain/src/main/groovy/life/qbic/business/offers/TaxOffice.groovy

```diff
@@ -0,0 +1,67 @@
+package life.qbic.business.offers
+
+import life.qbic.datamodel.dtos.business.Affiliation
+import life.qbic.datamodel.dtos.business.AffiliationCategory
+import life.qbic.datamodel.dtos.business.services.ExternalServiceProduct
+
+
+/**
+ * <b>Tax office that is responsible for offer tax calculation.</b>
+ *
+ * <p>Calculates the taxes that need to be considered for a given price value, affiliation and product type</p>
+ *
+ * <p>The current implementation takes a VAT ratio of 19%, default for Germany. Taxes only apply for customers within Germany
+ * and if they are not part of the University or UKT. Taxes are not forwarded internally.</p>
+ *
+ * <p>The only exception are products (services) external services, for which taxes need to
+ * be applied regardless of the affiliation status.</p>
+ *
+ * @since 1.2.0
+ */
+class TaxOffice {
+
+    private static final BigDecimal VAT_RATIO = BigDecimal.valueOf(0.19)
+
+    private final Affiliation affiliation
+
+    /**
+     * Creates a tax office that is configured by the affiliation.
+     *
+     * @param affiliation the affiliation the tax office instance needs to consider for calculations
+     */
+    TaxOffice(Affiliation affiliation) {
+        this.affiliation = affiliation
+    }
+
+    /**
+     * Calculates the absolute tax value that applies to a given service cost value and product type.
+     *
+     * @param serviceCosts the service cost value the office has to consider
+     * @param product the service product
+     * @return the absolute resulting tax value
+     */
+    BigDecimal applyTaxes(BigDecimal serviceCosts, Class clazz) {
```

Comment:
Why do you use the general Class class instead of the product class?

```suggestion
    BigDecimal applyTaxes(BigDecimal serviceCosts, Product product) {
```

Reply:
Actually I had that before, but decided to not pass the Product object with its information, because it is just not relevant for the function.

Reply:
I see. I think I would prefer product here because the context of this class is more restricted then with Class. I would expect to only obtain products that are of type product and this is not provided like this.